plugins/adduser/adduser.plugin.zsh

```diff
@@ -0,0 +1,55 @@
+# Copyright (c) 2021 Nikolas Garofil
+
+adduser() {
+	local returncode
+	local result_string="Installation of 'Oh My Zsh' "
+	local temp_installscript
+	local path_installscript="$ZSH/tools/install.sh"
+
+
+	#Create user, errors will be reported by the 'real' adduser
+	#Use which/tail combination to call the binary instead of this function
+	$(which -a adduser | tail -1) $@ || return 1
+
+
+	echo "\nUser '${@[$#]}' has been created. I will now try to install 'Oh My Zsh'"
+	if [[ -f $path_installscript ]] ; then
+
+		#copy install.sh to a new file in temp that we can give the right owner to execute
+		#and also make sure that after the install script we are no longer the new user
+		temp_installscript=$(mktemp)
+		cat $path_installscript | \
+		sed 's/exec zsh -l//' |
+	        sed 's/read -r opt/opt=y; echo "\n--- This time I am answering \\"yes\\" for you, but you will still have to type in the password of that user ---"/' \
```

Comment:
The install script allows bypassing certain steps. Look at the head section of the `install.sh` script for the options you want. I think you want to stop the final `exec`, and do the `chsh`.

The first can be skipped with `RUNZSH=no`, but the second can't unless you do what you did. In that case, I'd just run the script with `--unattended`, which skips both `exec zsh` and `chsh`, and do the `chsh` part in your call to `command adduser` (you can specify the shell with `adduser --shell <shell>`.

```sh
sh install.sh --unattended
```